nntrainer/layers/lstm.cpp

```diff
@@ -225,25 +213,16 @@ void LSTMLayer::forwarding(RunLayerContext &context, bool training) {
   Tensor &cs = context.getTensor(wt_idx[LSTMParams::cell_state]);
   Tensor &ifgos = context.getTensor(wt_idx[LSTMParams::ifgo]);
 
-  if (!start_timestep) {
-    hs.setZero();
-    cs.setZero();
-  }
-
-  /**
-   * @note when the recurrent realization happens, different instances of lstm
-   * will share the weights, hidden state, cell and ifgo memory. However, they
-   * do not share the input, output and derivatives memory. The input/output
-   * will be contain a single timestep data only.
-   */
+  hs.setZero();
+  cs.setZero();
 
   for (unsigned int batch = 0; batch < batch_size; ++batch) {
     const Tensor input_batch = inputs.getBatchSlice(batch, 1);
     Tensor hs_batch = hs.getBatchSlice(batch, 1);
     Tensor cs_batch = cs.getBatchSlice(batch, 1);
     Tensor ifgo_batch = ifgos.getBatchSlice(batch, 1);
 
-    for (unsigned int t = start_timestep; t < end_timestep; ++t) {
+    for (unsigned int t = 0; t < max_timestep; ++t) {
```

Comment:
I wonder time step always starts from 0?

Reply:
Yes. In the lstm layer timestep always starts from 0.